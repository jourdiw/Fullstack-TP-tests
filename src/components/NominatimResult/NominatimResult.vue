<template>
  <v-data-table
    :headers="headers"
    :items="result"
    class="elevation-1 ma-2"
    :loading="loading"
  >
    <template slot="items" slot-scope="props">
      <tr @mouseover="$emit('over', props.item)" @mouseout="$emit('out', props.item)">
        <td>
          <img v-if="props.item.icon" :src="props.item.icon" />
          <span class='grey--text'>
            {{ props.item.class }}
          </span>
          <br/>
          {{ props.item.display_name }}
        </td>
        <td>
          <v-btn flat icon color="primary" @click="$emit('select', props.item)">
            <v-icon>chevron_right</v-icon>
          </v-btn>
        </td>
      </tr>
    </template>
  </v-data-table>
</template>

<script>
export default {
  name: 'NominatimResult',
  props: ['result', 'loading'],
  data() {
    return {
      headers: [
        { text: 'Nom', align: 'left', value: 'display_name' },
      ],
    };
  },
};
</script>
