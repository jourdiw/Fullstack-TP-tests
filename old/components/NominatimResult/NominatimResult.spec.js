/* global describe it expect */
import { mount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import NominatimResult from './NominatimResult.vue';

const localVue = createLocalVue();
localVue.use(Vuetify);

describe('NominatimResult.spec.js', () => {
  it('is a Vue instance', () => {
    const vm1 = mount(NominatimResult, {
      localVue,
    });
    expect(vm1.isVueInstance()).toBeTruthy();
  });

  // it('has the right data to display', () => {
  //   const wrapper = mount(NominatimResult, {
  //     propsData: {
  //       result: 'Hello World',
  //     },
  //     localVue,
  //   });
  //   expect(wrapper.props().result).toBe('Hello World');
  // });

  it('has the expected html structure', () => {
    const vm3 = mount(NominatimResult, {
      propsData: {
        result: [{
          display_name: 'Hello World 2',
          class: 'pouet',
        }, {
          display_name: 'Hello World 3',
          class: 'pouet',
        }],
      },
      localVue,
    });
    expect(vm3.element).toMatchSnapshot();
  });
});
