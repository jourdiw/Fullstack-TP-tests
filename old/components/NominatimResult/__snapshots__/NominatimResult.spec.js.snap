// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NominatimResult.spec.js has the expected html structure 1`] = `
<div
  class="elevation-1 ma-2"
>
  <div
    class="table__overflow"
  >
    <table
      class="datatable table"
    >
      <thead>
        <tr>
          <th
            aria-label="Nom: Sorted ascending. Activate to sort descending."
            aria-sort="ascending"
            class="column sortable active asc text-xs-left"
            role="columnheader"
            scope="col"
            tabindex="0"
          >
            Nom
            <i
              aria-hidden="true"
              class="icon material-icons"
              style="font-size: 16px;"
            >
              arrow_upward
            </i>
          </th>
        </tr>
        <tr
          class="datatable__progress"
        >
          <th
            class="column"
            colspan="1"
          />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <!---->
             
            <span
              class="grey--text"
            >
              
          pouet
        
            </span>
             
            <br />
            
        Hello World 2
      
          </td>
           
          <td>
            <button
              class="btn btn--flat btn--icon primary--text"
              type="button"
            >
              <div
                class="btn__content"
              >
                <i
                  aria-hidden="true"
                  class="icon material-icons"
                >
                  chevron_right
                </i>
              </div>
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <!---->
             
            <span
              class="grey--text"
            >
              
          pouet
        
            </span>
             
            <br />
            
        Hello World 3
      
          </td>
           
          <td>
            <button
              class="btn btn--flat btn--icon primary--text"
              type="button"
            >
              <div
                class="btn__content"
              >
                <i
                  aria-hidden="true"
                  class="icon material-icons"
                >
                  chevron_right
                </i>
              </div>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="datatable table"
  >
    <div
      class="datatable__actions"
    >
      <div
        class="datatable__actions__select"
      >
        Rows per page:
        <div
          aria-label="Rows per page:"
          class="input-group input-group--append-icon input-group--hide-details input-group--text-field input-group--select input-group--auto primary--text"
          data-uid="19"
          role="combobox"
          tabindex="0"
        >
          <div
            class="input-group__input"
          >
            <div
              class="input-group__selections"
              style="overflow: hidden;"
            >
              <input
                aria-label="Rows per page:"
                class="input-group--select__autocomplete"
                disabled="disabled"
                style="display: none;"
                tabindex="-1"
              />
            </div>
            <div
              class="menu"
              style="display: inline-block;"
            >
              <div
                class="menu__content menu__content--select menu__content--auto"
                style="max-height: 200px; min-width: 75px; max-width: auto; top: 18px; left: 12px; z-index: 0; display: none;"
              >
                <div
                  class="card"
                  style="height: auto;"
                >
                  <div
                    class="list"
                  >
                    <div>
                      <a
                        class="list__tile list__tile--link"
                      >
                        <div
                          class="list__tile__content"
                        >
                          <div
                            class="list__tile__title"
                          >
                            5
                          </div>
                        </div>
                      </a>
                    </div>
                    <div>
                      <a
                        class="list__tile list__tile--link"
                      >
                        <div
                          class="list__tile__content"
                        >
                          <div
                            class="list__tile__title"
                          >
                            10
                          </div>
                        </div>
                      </a>
                    </div>
                    <div>
                      <a
                        class="list__tile list__tile--link"
                      >
                        <div
                          class="list__tile__content"
                        >
                          <div
                            class="list__tile__title"
                          >
                            25
                          </div>
                        </div>
                      </a>
                    </div>
                    <div>
                      <a
                        class="list__tile list__tile--link"
                      >
                        <div
                          class="list__tile__content"
                        >
                          <div
                            class="list__tile__title"
                          >
                            All
                          </div>
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <i
              aria-hidden="true"
              class="icon material-icons input-group__append-icon input-group__icon-cb"
            >
              arrow_drop_down
            </i>
          </div>
          <div
            class="input-group__details"
          />
        </div>
      </div>
      <div
        class="datatable__actions__range-controls"
      >
        <div
          class="datatable__actions__pagination"
        >
          1-2 of 2
        </div>
        <button
          aria-label="Previous page"
          class="btn btn--disabled btn--flat btn--icon"
          disabled="disabled"
          type="button"
        >
          <div
            class="btn__content"
          >
            <i
              aria-hidden="true"
              class="icon material-icons"
            >
              chevron_left
            </i>
          </div>
        </button>
        <button
          aria-label="Next page"
          class="btn btn--disabled btn--flat btn--icon"
          disabled="disabled"
          type="button"
        >
          <div
            class="btn__content"
          >
            <i
              aria-hidden="true"
              class="icon material-icons"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
